### PR TITLE
Issue 143

### PR DESCRIPTION
**Pull request by gst**
Fix issue #143 

Adapted `format_query_response()` to also handle tags returned.

I'd like some advices on the format I use to distinguish same serie with different tags.
Basically : if there are some tag(s) returned by a query then store them in the result by providing a unique entry for each such unique serie + tag(s) value(s) with `name = (row['name'], tuple(tags.values()))`

Now after thinking about it twice I think in fact I should also include the tags.keys() somewhere.. also more than probably in the `name` attribute of each result dict with `name = (row['name'], tuple(tags.items()))`   (*items* simply instead of *values* so) ; so that the caller can know which tag value correspond to which tag name..  otherwise if the groupby is done on multiple tags so then the caller can't know for sure the order of the tags as returned in the answer.

wdyt ?

### CODE REVIEW

**Comment by ReAzem** [comment]
I think that the series identifier should be a tuple with a name and a dict.

for example: ``("cpu", {"host_name": "example.com"})``

**Comment by gst** [comment]
> I think that the series identifier should be a tuple with a name and a dict.
> for example: ("cpu", {"host_name": "example.com"})

yep, that's what I also get in my second thought presented here above.

Problem is that the `name` attribute of the main dict result (from the `format_query_response` function) is a key (in the dict result so), so we can't use a dict as one of its component (not hashable).. so my `tags.items()` good second choice I think..

**Comment by ReAzem** [comment]
> so we can't use a dict as one of its component (not hashable)

You are right! :(

> so my tags.items() good second choice I think..

Absolutely.

**Comment by gst** [comment]
will travis make it succeed on the first pass ?

**Comment by ReAzem** [comment]
> will travis make it succeed on the first pass ?
What?

**Comment by gst** [comment]
> What?

I mean  : "is travis going to succeed the tests without any temporary error" :p

**Comment by ReAzem** [comment]
Looks like it failed for a valid reason. The order of tags is problematic. Looks like we will have to sort them.

**Comment by gst** [comment]
fixed in the test.

I'm not in favor of sorting things "undesirably" in the library itself. If one wants the sorted serie names from the library it should re-use the same "trick".. after its query() call. 

So I could eventually put the function `sorted_series_tags` in the library itself. (with a better name eventually ? if you have one..)  or not at all. 
what do you think ?

**Comment by ReAzem** [comment]
We are not sorting the series but just the tags,  so I don't see why this is underisable.

**Comment by gst** [comment]
Hmm, makes sense..

File: influxdb/client.py [hunk 1]
```diff
@@ -67,7 +69,9 @@ def __init__(self,
                  verify_ssl=False,
                  timeout=None,
                  use_udp=False,
-                 udp_port=4444):
+                 udp_port=4444,
+                 keep_json_response_order=True,
```
Comment (ReAzem) [inline influxdb/client.py]: Add it to the docstring please.

**Comment by gst** [comment]
Seems all good..

**Comment by ReAzem** [comment]
Seems good to me too!

**Pull request merged**